### PR TITLE
Add support for Node 12

### LINKED ISSUE #2632: support for node.js 12.x
node v12.0.0 just got released: https://nodejs.org/en/blog/release/v12.0.0/

installation is failing on macOS mojave

- NPM version (`npm -v`): 6.9.0
- Node version (`node -v`): 12.0.0
- Node Process (`node -p process.versions`):  [see below]
- Node Platform (`node -p process.platform`): darwin
- Node architecture (`node -p process.arch`): x64
- node-sass version (`node -p "require('node-sass').info"`): `Error: Node Sass does not yet support your current environment: OS X 64-bit with Unsupported runtime (72)`
- npm node-sass versions (`npm ls node-sass`): node-sass@4.11.0

```js
// process.versions:
{
  node: '12.0.0',
  v8: '7.4.288.21-node.16',
  uv: '1.28.0',
  zlib: '1.2.11',
  brotli: '1.0.7',
  ares: '1.15.0',
  modules: '72',
  nghttp2: '1.38.0',
  napi: '4',
  llhttp: '1.1.1',
  http_parser: '2.8.0',
  openssl: '1.1.1b',
  cldr: '34.0',
  icu: '63.1',
  tz: '2018e',
  unicode: '11.0'
}
```

### PR DESCRIPTION
**Pull request by xzyfer**
Fixes #2632

### CODE REVIEW

**Comment by xzyfer** [comment]
Looks like https://github.com/nodejs/nan doesn't yet support Node 12 so we're blocked on compiling Node 12 binaries.

**Comment by xzyfer** [comment]
Created a tracking issue https://github.com/nodejs/nan/issues/849

**Comment by nschonni** [comment]
Also will need https://github.com/appveyor/ci/issues/2921

File: appveyor.yml [hunk 1]
```diff
@@ -73,6 +73,9 @@
       - nodejs_version: 11
         GYP_MSVS_VERSION: 2015
         APPVEYOR_BUILD_WORKER_IMAGE: Visual Studio 2015
+      - nodejs_version: 12
+        GYP_MSVS_VERSION: 2015
```
Comment (nschonni) [inline appveyor.yml]: Think this and 10/11 are actually supposed to be 2017 https://github.com/nodejs/build/blob/639b55d6e7efb911898c2ada97d0b4099ee86d4f/doc/windows-visualstudio-supported-versions.md#for-building-nodejs-core

Comment (xzyfer) [reply appveyor.yml]: Updated.

File: .travis.yml [hunk 2]
```diff
@@ -17,6 +17,9 @@ jobs:
     - stage: platform-test
       node_js: "node"
       os: osx
+    - stage: platform-test
+      node_js: "11"
+      os: linux
```
Comment (nschonni) [inline .travis.yml]: Probably need an explicit OSX 11 too

File: .travis.yml [hunk 3]
```diff
@@ -17,6 +17,9 @@ jobs:
...
     - stage: platform-test
       node_js: "10"
```
Comment (nschonni) [inline .travis.yml]: Could probably flip the 10s over to `lts/dubnium` now. Not sure if that makes it clearer or not, but I was just seeing that we/I did that for the other LTS in the matrix

File: .travis.yml (hunk 2 above)
Comment (xzyfer) [reply .travis.yml]: Updated.

File: .travis.yml (hunk 3 above)
Comment (xzyfer) [reply .travis.yml]: I'm finding the lts labels confusing now. I've switched them all to be version numbers instead.

**Comment by xzyfer** [comment]
Got this green using the workaround from https://github.com/appveyor/ci/issues/2921#issuecomment-486727727. Hopefully this should mean no more waiting for AppVeyor.

**Comment by nschonni** [comment]
Might want to update the package.json with this, unless you are going to do that separately before tagging

**Comment by xzyfer** [comment]
Bumped package.json on master

**Pull request merged**